crypto/ec/ecdsa_sign.c

```diff
@@ -20,6 +20,10 @@ ECDSA_SIG *ECDSA_do_sign_ex(const unsigned char *dgst, int dlen,
                             const BIGNUM *kinv, const BIGNUM *rp,
                             EC_KEY *eckey)
 {
+    int curvebits = EC_GROUP_order_bits(EC_KEY_get0_group(eckey));
+    if (curvebits && dlen > (curvebits + 7) / 8)
+        dlen = (curvebits + 7) / 8;
```

Comment:
This is incomplete we do quite a bit more:
https://github.com/openssl/openssl/blob/f42c225d7f9a0bce0bf46103343402d3f0ad742f/crypto/ec/ecdsa_ossl.c#L242-L252